### ISSUE TITLE
sysrc module can't set OID-like variables

### ISSUE THREAD

**Issue by b-a-t** [dlundgren/ansible-freebsd-modules#11]
OID-like variables are the variables like `vm.kmem_size` or `vfs.zfs.vdev.cache.size`, which are used in `/boot/loader.conf` or `/etc/sysctl.conf` for example.

Editing them with `sysrc`used to work in the past, but since mid-summer 2018 a patch from [PR 187461](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=187461) was applied and things broke.

Now, on an attempt to use such a variable an error is displayed:
```
# sysrc -f /boot/loader.conf vm.kmem_size
sysrc: vm.kmem_size: name contains characters not allowed in shell
```
and an exit code of 1 is returned, which is not detected by the `sysrc` module:
```
TASK [zfs : ensure /boot/loader.conf has correct the ZFS settings] *********************************************************************************************************************************************************************************
ok: [10.168.170.3] => (item={u'name': u'zfs_load', u'value': u'YES'}) => {"changed": false, "delim": " ", "dest": "/boot/loader.conf", "gid": 0, "group": "wheel", "item": {"name": "zfs_load", "value": "YES"}, "mode": "0644", "name": "zfs_load", "owner": "root", "size": 87, "state": "file", "uid": 0, "value": "YES"}
ok: [10.168.170.3] => (item={u'name': u'vm.kmem_size', u'value': u'330M'}) => {"changed": false, "delim": " ", "dest": "/boot/loader.conf", "gid": 0, "group": "wheel", "item": {"name": "vm.kmem_size", "value": "330M"}, "mode": "0644", "name": "vm.kmem_size", "owner": "root", "size": 87, "state": "file", "uid": 0, "value": "330M"}
ok: [10.168.170.3] => (item={u'name': u'vm.kmem_size_max', u'value': u'330M'}) => {"changed": false, "delim": " ", "dest": "/boot/loader.conf", "gid": 0, "group": "wheel", "item": {"name": "vm.kmem_size_max", "value": "330M"}, "mode": "0644", "name": "vm.kmem_size_max", "owner": "root", "size": 87, "state": "file", "uid": 0, "value": "330M"}
ok: [10.168.170.3] => (item={u'name': u'vm.zfs.arc_max', u'value': u'40M'}) => {"changed": false, "delim": " ", "dest": "/boot/loader.conf", "gid": 0, "group": "wheel", "item": {"name": "vm.zfs.arc_max", "value": "40M"}, "mode": "0644", "name": "vm.zfs.arc_max", "owner": "root", "size": 87, "state": "file", "uid": 0, "value": "40M"}
ok: [10.168.170.3] => (item={u'name': u'vfs.zfs.vdev.cache.size', u'value': u'5M'}) => {"changed": false, "delim": " ", "dest": "/boot/loader.conf", "gid": 0, "group": "wheel", "item": {"name": "vfs.zfs.vdev.cache.size", "value": "5M"}, "mode": "0644", "name": "vfs.zfs.vdev.cache.size", "owner": "root", "size": 87, "state": "file", "uid": 0, "value": "5M"}

PLAY RECAP *****************************************************************************************************************************************************************************************************************************************
10.168.170.3               : ok=4    changed=0    unreachable=0    failed=0
```

**Comment by dlundgren** [comment]
Although it never made it in to the readme, there is a note in the documentation for the `sysrc` module that periods aren't allowed, due to the shell restrictions.

That being said, I think this module should allow them, since some files don't have these restrictions. I'm looking in to it.

**Comment by b-a-t** [comment]
The problem is not in the (ansible) module `sysrc`, but in the system `sysrc` shell script.
It used to work in the past, but since mid-summer it got broken(or fixed).

For your module that means that it has to catch that new error message and exit code, when system script fails due to the use of variable with the dots in it.

I think that `sysrc` module should fail in such case as well, so users won't get confused by that operation succeeded, but there are no changes in the target file. For me it took few hours to realize that something is wrong and to find out, why.

In addition, it would be great that on, lets say, *-vvvvv* verbosity level you can see what shell command was issued(and, maybe, it's output as well).

Sorry for not providing patches...

**Comment by dlundgren** [comment]
I have #13 up for review, which basically just checks that the `name` falls within the same range of valid characters that `sysrc` likes. This also has the verbosity level idea incorporated as well.

**Comment by dlundgren** [comment]
I merged in #13 to fix this value checking.
